### PR TITLE
Add tsp-client local spec development docs

### PR DESCRIPTION
**Pull request by catalinaperalta**
Adding docs to explain the local development scenario for folks who havent checked a spec in yet or who want to work on local changes and generate a client library based on those updates.

### CODE REVIEW

File: website/src/content/docs/docs/howtos/Generate with tsp-client/local_spec_quickstart.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+title:  Quickstart - Local development
+---
+
+`tsp-client` allows you to develop your TypeSpec project locally before pushing changes upstream. These instructions can be used when you want to generate a client library from a local TypeSpec project that has the changes you wish to use.
+
+### Prerequisites
+
+- Install `tsp-client`:
+
+```pwsh
```
Comment (timotheeguerin) [inline website/src/content/docs/docs/howtos/Generate with tsp-client/local_spec_quickstart.md:11]:
```suggestion
```bash
```

Comment (timotheeguerin) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/local_spec_quickstart.md:11]: same for the others

File: website/src/content/docs/docs/howtos/Generate with tsp-client/local_spec_quickstart.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+---
+title:  Quickstart - Local development
```
Comment (timotheeguerin) [inline website/src/content/docs/docs/howtos/Generate with tsp-client/local_spec_quickstart.md:2]: Wondering if this name makes sense right next to `tsp-client usage`

should we name them differently?

Comment (catalinaperalta) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/local_spec_quickstart.md:2]: I agree...Maybe it should be:
- `tsp-client command documentation`
- `tsp-client quickstart`

OR maybe just remove tsp-client and do:
- `Supported commands`
- `Quickstart (local development)`

Comment (catalinaperalta) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/local_spec_quickstart.md]: I liked the second option more so updated to that, let me know if you have a better idea!

File: website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
-title: tsp-client usage
+title: Supported commands
```
Comment (timotheeguerin) [inline website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md:2]:
```suggestion
title: Commands
```

File: website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
...
 ---
```
Comment (timotheeguerin) [inline website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md:4]: also wondering if you should hjave the install step as a separate page as you need it for both docs

File: website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md (hunk 3 above)
Comment (timotheeguerin) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md:2]: actually thats not correct, it does more than show the commands, 
image

File: website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md (hunk 4 above)
Comment (catalinaperalta) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md:4]: In my experience folks dont like to really read through multiple pages of docs, so I'd rather they have all the steps they need on the same page. We can look at this again in the future if there are more shared instructions between both pages.

File: website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
...
-title: tsp-client usage
+title: Commands
```
Comment (timotheeguerin) [inline website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md:2]: Sorry reopening as my reply might have got lost, I think actually this name is not really accurate 
image

I think either we should split this file or maybe find a better name

File: website/src/content/docs/docs/howtos/Generate with tsp-client/local_spec_quickstart.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
+---
+title: Quickstart (local development)
```
Comment (timotheeguerin) [inline website/src/content/docs/docs/howtos/Generate with tsp-client/local_spec_quickstart.md:2]: what would be the other type of developemnt?

File: website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md (hunk 5 above)
Comment (catalinaperalta) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md:2]: Ok I dont want to do the refactoring for the other doc in this PR, but I agree I think we can find a way to split it. Maybe between usage & configuration (dev), commands and configurations, configuration (repo). 

For this PR I could revert the renaming we just did or update it to `Usage & configuration` which I think covers most of the content of the current documentation, what do you think?

File: website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md (hunk 4 above)
Comment (catalinaperalta) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md:4]: Sorry I thought this comment was on the quickstart page...in my comment below I agree to splitting up this doc into separate sections

File: website/src/content/docs/docs/howtos/Generate with tsp-client/local_spec_quickstart.md (hunk 6 above)
Comment (timotheeguerin) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/local_spec_quickstart.md:2]: yeah I think it make sense as just QuickStart

File: website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md (hunk 5 above)
Comment (timotheeguerin) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md:2]: however you feel, just depends I think on when the follow up will be done

Comment (timotheeguerin) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md:2]: `Usage & Configuration` dos also seems like a good name without needing a split

Comment (timotheeguerin) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md:2]:
```
tsp-client
  Quick Start
  Usage & Configuration
```

**Review by timotheeguerin — APPROVED** [review]
Looks good I think with the change discussed in the comments

File: website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md (hunk 5 above)
Comment (catalinaperalta) [reply website/src/content/docs/docs/howtos/Generate with tsp-client/tsp_client.md:2]: Agree will update to those names for now, and I opened this tracking issue to look into splitting the docs: https://github.com/Azure/azure-sdk-tools/issues/9357

**Pull request merged**